src/cargo/sources/git/utils.rs

```diff
@@ -1084,6 +1086,28 @@ fn fetch_with_cli(
     } else {
         cmd.arg("--no-tags");
     }
+
+    match shallow {
```

Comment:
I don't think we are using variants other than `NoChange`/`DepthAtRemote`. We might just leave then `unimplmented!(…)`. 

https://github.com/rust-lang/cargo/blob/611b7c4ee40b63d0818040f7290a8e14424a9519/src/cargo/sources/git/mod.rs?plain=1#L35-L57